### DISCUSSION TITLE
While running tests, it takes entire system resources.

### DISCUSSION THREAD

**Discussion by nishu-murmu** [wxt-dev/wxt#1250] [General]
I was working on some e2e testing.
I noticed that at my work PC, I was using Windows, while the tests were running, my vscode would get killed multiple times.
Then I did some work at home PC, where I used Linux, I noticed, it was taking almost entire resources.

![2024-12-05_23-38](https://github.com/user-attachments/assets/3079e166-00c5-48e3-a546-64d37c088866)

@aklinker1 Any insights on this?

**Comment by aklinker1** [comment]
🤷 Probably could reduce the number of Vitest's workers. Looks like by default, it spawns the same number of child processes as you have CPU cores. Reducing this number would likely reduce memory consumption as well. 

https://vitest.dev/config/#pooloptions-forks-maxforks

My suggestion would be to just run a single test if possible. Otherwise, if I need to run all tests, I want them to get done, so using all my CPU cores is fine. Just run all of them before opening a PR or wait for a maintainer to trigger CI to see full test results.

**Reply by nishu-murmu** [reply]
Running single tests is the way for now. Thanks.
